src/main/java/com/snowflake/kafka/connector/internal/SnowflakeIngestionServiceV1.java

```diff
@@ -254,4 +257,104 @@ public void close() {
   public SimpleIngestManager getIngestManager() {
     return this.ingestManager;
   }
+
+  /**
+   * configure the Snowpipe client and return the client sequencer
+   *
+   * @return a Long value contains the client sequencer
+   */
+  public Long configureClient() {
+    ConfigureClientResponse response;
+    try {
+      response =
+          (ConfigureClientResponse)
+              InternalUtils.backoffAndRetry(
+                  telemetry,
+                  SnowflakeInternalOperations.CONFIGURE_CLIENT_SNOWPIPE_API,
+                  () -> ingestManager.configureClient(null));
+    } catch (Exception e) {
+      throw SnowflakeErrors.ERROR_3006.getException(e);
+    }
+    if (response != null && response.getClientSequencer() != null) {
+      return response.getClientSequencer();
+    } else {
+      throw SnowflakeErrors.ERROR_4001.getException("the response of configure client is null ");
+    }
+  }
+
+  /**
+   * get the Snowpipe client and return the offset token
+   *
+   * @return a String value contains the offset token
+   */
+  public String getClientStatus() {
+    ClientStatusResponse response;
```

Comment:
should we pass the clientSequencer and make sure it matches with returned clientSequencer?

Reply:
If we are not passing it in API call, we can avoid doing it in this method. 
This method can be thought of as a wrapper on top of remote API. Lets avoid adding more checks. 

May be that assert can be done by the caller?